### PR TITLE
 Reorder some of the flow in CompilerStack::compileContract for readability

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: libsolidity/interface/CompilerStack.h [hunk 1]
```diff
@@ -273,9 +273,9 @@ class CompilerStack: boost::noncopyable
 	std::string applyRemapping(std::string const& _path, std::string const& _context);
 	void resolveImports();
 	/// @returns the absolute path corresponding to @a _path relative to @a _reference.
-	std::string absolutePath(std::string const& _path, std::string const& _reference) const;
+	static std::string absolutePath(std::string const& _path, std::string const& _reference);
 	/// Helper function to return path converted strings.
-	std::string sanitizePath(std::string const& _path) const { return boost::filesystem::path(_path).generic_string(); }
+	static std::string sanitizePath(std::string const& _path) { return boost::filesystem::path(_path).generic_string(); }
```
Comment (axic) [inline libsolidity/interface/CompilerStack.h:278]: Actually these two could be moved to `libdevcore`.

File: libsolidity/interface/CompilerStack.cpp [hunk 2]
```diff
@@ -711,9 +711,15 @@ void CompilerStack::compileContract(
 	for (auto const* dependency: _contract.annotation().contractDependencies)
 		compileContract(*dependency, _compiledContracts);
 
-	shared_ptr<Compiler> compiler = make_shared<Compiler>(m_evmVersion, m_optimize, m_optimizeRuns);
 	Contract& compiledContract = m_contracts.at(_contract.fullyQualifiedName());
+
+	shared_ptr<Compiler> compiler = make_shared<Compiler>(m_evmVersion, m_optimize, m_optimizeRuns);
```
Comment (chriseth) [inline libsolidity/interface/CompilerStack.cpp:716]: Perhaps it is better to not introduce this local variable at all? Same goes for `metadata` below.

**Review by chriseth — APPROVED** [review]

File: libsolidity/interface/CompilerStack.cpp (hunk 2 above)
Comment (axic) [reply libsolidity/interface/CompilerStack.cpp:716]: `metadata` is used twice: here and for hashing. Though it could use `compiledContract.metadata`.

Comment (axic) [reply libsolidity/interface/CompilerStack.cpp:716]: `compiler` is used at least 3 times, it probably makes sense to have the local variable for readability.

**Comment by axic** [comment]
Merging this for now, we can still improve it after.

**Pull request merged**